### PR TITLE
Resource module for IOS ACL_Interfaces

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Resource module for IOS ACL_Interfaces. Resource model ref: [ios_acl_interfaces](https://github.com/ansible-network/resource_module_models/pull/58)

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

ios_acl_interfaces

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

File: lib/ansible/module_utils/network/ios/config/acl_interfaces/acl_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.network.common.cfg.base import ConfigBase
```
Comment (NilashishC) [inline lib/ansible/module_utils/network/ios/config/acl_interfaces/acl_interfaces.py:16]: We probably need to use normalize_interfaces method in this module.

File: lib/ansible/modules/network/ios/ios_acl_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+# interface GigabitEthernet0/0
+# interface GigabitEthernet0/1
+#  ip access-group 110 in
+#  ip access-group 123 out
+#  ipv6 traffic-filter test_v6 out
+#  ipv6 traffic-filter temp_v6 in
+# interface GigabitEthernet0/2
+#  ip access-group 110 in
+#  ip access-group 123 out
+
```
Comment (NilashishC) [inline lib/ansible/modules/network/ios/ios_acl_interfaces.py:178]: Please add an example for updating existing attributes using merged.

File: test/integration/targets/ios_acl_interfaces/tests/cli/merged.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+---
```
Comment (NilashishC) [inline test/integration/targets/ios_acl_interfaces/tests/cli/merged.yaml:1]: Please add assertions for before/after dicts in all the test.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/ios/ios_acl_interfaces.py (hunk 2 above)
Comment (justjais) [reply lib/ansible/modules/network/ios/ios_acl_interfaces.py:178]: +1

File: lib/ansible/module_utils/network/ios/config/acl_interfaces/acl_interfaces.py (hunk 1 above)
Comment (justjais) [reply lib/ansible/module_utils/network/ios/config/acl_interfaces/acl_interfaces.py:16]: not needed

File: test/integration/targets/ios_acl_interfaces/tests/cli/merged.yaml (hunk 3 above)
Comment (justjais) [reply test/integration/targets/ios_acl_interfaces/tests/cli/merged.yaml:1]: IMO it's not required to specifically check for before and after dicts generated if facts is working as expected and commands fired are correct.

**Comment by justjais** [comment]
recheck

File: lib/ansible/modules/network/ios/ios_acl_interfaces.py [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+                  - out
+  running_config:
+    description:
+      - The module, by default, will connect to the remote device and retrieve the current
+        running-config to use as a base for comparing against the contents of source.
+        There are times when it is not desirable to have the task get the current running-config
+        for every task in a playbook.  The I(running_config) argument allows the implementer to
+        pass in the configuration to use as the base config for comparison. This value of this
+        option should be the output received from device by executing command.
+        C(show configuration commands | grep 'static route')
```
Comment (NilashishC) [inline lib/ansible/modules/network/ios/ios_acl_interfaces.py]: The command needs to be updated.

**Review by NilashishC — APPROVED** [review]

**Pull request merged**